code/modules/keybindings/bindings_client.dm

```diff
@@ -55,20 +75,22 @@
 			keys_held[i] = null
 			break
 
-	var/movement = SSinput.movement_keys[_key]
+	//Can't just do a remove because it would alter the length of the rolling buffer, instead search for the key then null it out if it exists
+	for(var/i in 1 to HELD_KEY_BUFFER_LENGTH)
+		if(keys_held[i] == _key)
+			keys_held[i] = null
+			break
+	var/movement = movement_keys[_key]
 	if(!(next_move_dir_add & movement))
 		next_move_dir_sub |= movement
 
 	// We don't do full key for release, because for mod keys you
 	// can hold different keys and releasing any should be handled by the key binding specifically
 	for (var/kb_name in prefs.key_bindings[_key])
 		var/datum/keybinding/kb = GLOB.keybindings_by_name[kb_name]
-		if (kb.up(src))
+		if(kb.up(src))
 			break
 
-	holder?.key_up(_key, src)
-	mob.focus?.key_up(_key, src)
```

Comment:
Are you sure this isn't needed?

Reply:
Yes, these are being replaced with keybinds.